compiler/resolution/lifetime.cpp

```diff
@@ -1112,7 +1523,7 @@ static bool isLocalVariable(FnSymbol* fn, Symbol* sym) {
   return sym->defPoint->getFunction() == fn && isLocal;
 }
 
-static bool typeHasInfiniteLifetime(Type* type) {
+static bool typeHasInfiniteBorrowLifetime(Type* type) {
```

Comment:
Would be nice to know how "borrow" lifetime differs from "intrinsic" and "inferred" lifetimes.

Reply:
I'll add a comment but each "lifetime" (intrinsic or inferred) contains two components in the analysis - ref lifetime and borrow lifetime. One is for handling "borrow checking" the other one validates refs to locals do not escape.